auth/auth_handler.go

```diff
@@ -36,47 +36,51 @@ const (
 	errBody = `{"code": %d, "message": "%s"}`
 )
 
-// PortalAppStore interface provides an in-memory store of PortalApps.
+// portalAppStore interface provides an in-memory store of PortalApps.
 //
 // Used for:
-// - Fast lookups of authorization data for PATH when processing requests.
-type PortalAppStore interface {
+//   - Fast lookups of authorization data for PATH when processing requests.
+type portalAppStore interface {
 	GetPortalApp(portalAppID store.PortalAppID) (*store.PortalApp, bool)
 }
 
+// rateLimitStore interface provides an in-memory store of rate limited accounts.
+//
+// Used for:
+//   - Fast lookups of rate limited accounts for PATH when processing requests.
+type rateLimitStore interface {
+	IsAccountRateLimited(accountID store.AccountID) bool
+}
+
 // AuthHandler processes requests from Envoy.
 //
 // Primary responsibilities:
-// - Handles requests via the Check method (called for each request)
+//   - Handles requests via the Check method (called for each request)
 type AuthHandler struct {
 	Logger polylog.Logger
 
 	// PortalAppStore: in-memory store of PortalApps
-	PortalAppStore PortalAppStore
+	PortalAppStore portalAppStore
+
+	// RateLimitStore: in-memory store of rate limited accounts
+	RateLimitStore rateLimitStore
```

Comment:
Public field of private type seems inconsistent: please either make the interface public (e.g. if a user should be able to use any store), or make the field private (if the interface is there to facilitate unit testing).